### PR TITLE
Developer - Organize build targets into folders for IDEs

### LINKED ISSUE #7802: OPENGL_REQUIRED changes target folders in Visual Studio solution explorer
Issue overview
--------------
On the `transition_ems_func` branch with latest develop merged in, I configured without OpenGL and the fortran projects (ExpandObjects_build, and Transition_build) are present in the solution.

Check OPENGL_REQUIRED (BUILD_FORTRAN is still checked), reconfigure, and those project are now missing from the solution.

image  image

### Details
Some additional details for this issue (if relevant):
 - Win10, VS2019 
 - Version of EnergyPlus 9.3.0-a99bd2c44e

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by nealkruis**
Pull request overview
---------------------
 - Fixes #7802
 
This sets the [FOLDER property](https://cmake.org/cmake/help/latest/prop_tgt/FOLDER.html#prop_tgt:FOLDER) for all targets in the EnergyPlus build process so they are more sanely organized in IDEs. 

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)

### CODE REVIEW

**Comment by Myoldmopar** [comment]
Checked this out locally and found the makefile based build on command line and in CLion are untouched, which is totally fine.  I will not review these changes as I don't use Visual Studio.  I would encourage @mjwitte or @rraustad to test it out.  @jasondegraw please mark as approved or request changes as you have already brought up your concerns.  

The code is currently in BigLadder's fork of EnergyPlus, so you'll need to clone that locally (or add another remote to your clone)  (or other ways that aren't worth discussing here).

File: third_party/re2/CMakeLists.txt [hunk 1]
```diff
@@ -10,7 +10,7 @@ if(POLICY CMP0048)
 endif()
 
 project(RE2 CXX)
-include(CTest)
+#include(CTest)
```
Comment (mjwitte) [inline third_party/re2/CMakeLists.txt:13]: Just curious, why this change?

**Review by mjwitte — APPROVED** [review]
@nealkruis Pulled and tested locally in VS2019 16.6.1. Looks good. Also tested in VS2017, ~didn't have any impact on the solution explorer - probably because OPENGL_REQUIRED wasn't checked in that folder?~

File: third_party/re2/CMakeLists.txt (hunk 1 above)
Comment (nealkruis) [reply third_party/re2/CMakeLists.txt:13]: This is what adds the "CTestDashboardTargets" ("Continuous", "Experimental", "Nightly", "NightlyMemoryCheck") to appear. It is not actually needed to make tests or run CTest. It might be worth suggesting they remove it from re2.

**Review by nealkruis — COMMENTED** [review]
It should work in VS2017. OPENGL_REQUIRED shouldn't have any effect any more. Did you re-run CMake (or ZERO_CHECK) in that IDE?

**Comment by mjwitte** [comment]
@nealkruis Never mind. It helps to check out the correct branch first.

**Comment by Myoldmopar** [comment]
@jasondegraw if you want to make any more tweaks, comment as needed.  Otherwise we'll put this on the merge train, departing this afternoon for `develop` station.

**Comment by jasondegraw** [comment]
@Myoldmopar Every computer I have that can build E+ is currently doing so (on different branches), so I won't be able to test it myself. Don't hold up the train on my account.

**Comment by Myoldmopar** [comment]
@jasondegraw thanks for letting me know!  @mjwitte merge at will.  Thanks @nealkruis.
